### PR TITLE
Adding gulp tasks for generating jsdocs -> markdown API documentation.

### PR DESCRIPTION
**Pull request by fokusferit**
Adding gulp tasks for creating jsdocs -> markdown API documentation.
Basically:
- I'm building temporarily all components to ES2015 + comments
- Using JsdocsTool to generate Markdown out of jsdocs
- Saving them in dist/docs
- It is part of build process.

Will improve it, interested people can already check it out. 

- [x] Only create doc files for components (not import.js and index.js)

### CODE REVIEW

**Comment by fokusferit** [comment]
For now there is still an issue with generating docs:

-> As the transpiled modules are Objects jsdocs has some issues correctly understanding it.

**Comment by fokusferit** [comment]
FYI: I'm not adding this doc coverage tool for now because it is not supporting ES6 out of the box. So I will do a future PR for that process.

File: build/tasks/docs.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const files = glob.sync(paths.source); // glob allows pattern matching for filenames
+  const createdMdDocs = [];
+  files
+  .filter(filename => {
+    const name = filename.match(/((\w)+(\-)*(\w))+(?=.js)/g)[0];
+    return name !== 'imports';
+  })
+  .map(filename => {
+    const splittedName = filename.split('/');
+    const filenameWithoutExt = filename.match(/((\w)+(\-)*(\w))+(?=.js)/g)[0];
```
Comment (fragsalat) [inline build/tasks/docs.js]: You could make a constant for this regex

File: package.json [hunk 2]
```diff
@@ -94,5 +98,9 @@
   "peerDependencies": {
     "preact": "7.1.0",
     "preact-compat": "3.13.1"
+  },
+  "dependencies": {
```
Comment (fragsalat) [inline package.json]: Pls remove this. We don't want react or preact as runtime dependencies.

File: package.json [hunk 3]
```diff
@@ -61,11 +61,14 @@
     "chai": "3.5.0",
     "conventional-changelog": "^1.1.0",
     "css-loader": "0.26.1",
+    "documentation": "^4.0.0-beta.18",
```
Comment (fragsalat) [inline package.json]: There is no stable version?

File: build/tasks/build.js [hunk 4]
```diff
@@ -1,51 +1,51 @@
...
 });
 
-gulp.task('build-css', function() {
+gulp.task('build-css', () => {
   return gulp.src(paths.css)
-    .pipe(gulp.dest(paths.output + 'es2015'))
-    .pipe(gulp.dest(paths.output + 'commonjs'))
-    .pipe(gulp.dest(paths.output + 'amd'))
-    .pipe(gulp.dest(paths.output + 'system'));
+    .pipe(gulp.dest(`${paths.output  }es2015`))
```
Comment (fokusferit) [inline build/tasks/build.js]: idk why this happened -> might be something with my visual code auto-fix thing.

File: build/tasks/docs.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+const paths = require('../paths');
+const fs = require('fs-then-native');
+const glob = require('glob');
+const del = require('del');
+const vinylPaths = require('vinyl-paths');
+const docjs = require('documentation');
+
+// deletes all files in the output path
+gulp.task('clean-docs', () =>
+  gulp.src([`${paths.output}/docs/`])
```
Comment (fokusferit) [inline build/tasks/docs.js]: 👍  you're right. Will create docs/api for that

File: package.json (hunk 2 above)
Comment (fokusferit) [inline package.json]: 👍

File: package.json (hunk 3 above)
Comment (fokusferit) [inline package.json]: They have a v3.0.5 which is from 2015, it looks like they have moved to v4.0.0 but didn't decided yet to introduce one stable version for that. Maybe I should open an issue ^^

**Pull request merged**